files/scripts/asic_status.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+"""
+    bootstrap-asic
+"""
+try:
+    import re
+    import syslog
+    import sys
+    import threading
+    from sonic_py_common import daemon_base
+    from swsscommon import swsscommon
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+#
+# Constants ====================================================================
+#
+CHASSIS_ASIC_INFO_TABLE = 'CHASSIS_ASIC_TABLE'
+SELECT_TIMEOUT_MSECS = 5000
+
+def main():
+    stop_event = threading.Event()
+    if len(sys.argv) != 2:
+        raise Exception('Pass valid asic-id as argument')
+
+    args_asic_id = sys.argv[1]
+
+    # Get platform from machine.conf
+    platform='unknown'
+    with open('/host/machine.conf') as f:
+        for line in f.readlines():
+            k, v = line.rstrip("\n").split("=")
+            if (k == 'onie_platform') or (k == 'aboot_platform'):
+                platform=v
+                break
+
+    if platform == 'unknown':
+        raise Exception('Unable to detect valid platform')
+
+    # Get num asics
+    num_asics=0
+    asic_file = '/usr/share/sonic/device/'+platform+'/asic.conf'
+    with open(asic_file) as f:
+        for line in f.readlines():
+            k, v = line.rstrip("\n").split("=")
+            if (k == 'NUM_ASIC'):
+                num_asics=v
+                break
+
+    if num_asics == 0:
+        syslog.syslog(syslog.LOG_ERR,
+                'Detected no asics on this platform')
+        sys.exit(1)
+
+    # Connect to STATE_DB and subscribe to chassis-module table notifications
+    state_db = daemon_base.db_connect("STATE_DB")
+
+    sel = swsscommon.Select()
+    sst = swsscommon.SubscriberStateTable(state_db, CHASSIS_ASIC_INFO_TABLE)
+    sel.addSelectable(sst)
+
+    while not stop_event.is_set():
+        (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
```

Comment:
If There are max 32 asics and only 16 are inserted, there will be 16*4 service scripts running in this loop. Shouldn't we be doing async wait?

Reply:
It is async. It will wake up every 5 seconds and then again go back to select().